extractor/filesystem/embeddedfs/vmdk/vmdk_test.go

```diff
@@ -0,0 +1,185 @@
+		})
+	}
+}
+func TestExtractValidVMDK(t *testing.T) {
+	extractor := vmdk.New()
+	path := "testdata/valid.vmdk"
+	info, err := os.Stat(path)
+	if err != nil {
+		t.Fatalf("os.Stat(%q) failed: %v", path, err)
+	}
+	input := &filesystem.ScanInput{
+		Path:   path,
+		Root:   ".",
+		Info:   info,
+		Reader: nil,
+		FS:     nil,
+	}
+
+	ctx := context.Background()
+	inv, err := extractor.Extract(ctx, input)
+	if err != nil {
+		t.Fatalf("Extract(%q) failed: %v", path, err)
+	}
+
+	if len(inv.EmbeddedFSs) == 0 {
+		t.Fatal("Extract returned no DiskImages")
+	}
+
+	for i, embeddedFS := range inv.EmbeddedFSs {
+		t.Run(fmt.Sprintf("DiskImage_%d", i), func(t *testing.T) {
+			if !strings.HasPrefix(embeddedFS.Path, path) {
+				t.Errorf("EmbeddedFS.Path = %q, want prefix %q", embeddedFS.Path, path)
+			}
+
+			fs, _ := embeddedFS.GetEmbeddedFS(ctx)
+
+			entries, err := fs.ReadDir("/")
+			if err != nil {
+				t.Fatalf("fs.ReadDir(/) failed: %v", err)
+			}
+			t.Logf("ReadDir(/) returned %d entries", len(entries))
+
+			info, err := fs.Stat("/")
+			if err != nil {
+				t.Fatalf("fs.Stat(/) failed: %v", err)
+			}
+			if !info.IsDir() {
+				t.Errorf("fs.Stat(/) IsDir() = %v, want true", info.IsDir())
+			}
+
+			found := false
+			for _, entry := range entries {
+				name := entry.Name()
+				if strings.HasSuffix(name, ".pem") {
+					found = true
+					filePath := "/" + name
+					f, err := fs.Open(filePath)
+					if err != nil {
+						t.Fatalf("fs.Open(%q) failed: %v", filePath, err)
+					}
+					defer f.Close()
+
+					buf := make([]byte, 4096)
+					n, err := f.Read(buf)
+					if err != nil && !errors.Is(err, io.EOF) {
+						t.Errorf("f.Read(%q) failed: %v", filePath, err)
+					}
+					t.Logf("Read %d bytes from %s\n%s", n, name, buf)
+
+					info, err := f.Stat()
+					if err != nil {
+						t.Errorf("f.Stat(%q) failed: %v", filePath, err)
+					} else if info.IsDir() {
+						t.Errorf("f.Stat(%q) IsDir() = %v, want false", filePath, info.IsDir())
+					}
+					break
+				}
+			}
+			if !found {
+				t.Errorf("private keys not found")
+			}
+		})
+	}
+}
+
+func TestExtractInvalidVMDK(t *testing.T) {
+	extractor := vmdk.New()
+	path := "testdata/invalid.vmdk"
+	info, err := os.Stat(path)
+	if err != nil {
+		t.Fatalf("os.Stat(%q) failed: %v", path, err)
+	}
+	input := &filesystem.ScanInput{
+		Path:   path,
+		Root:   ".",
+		Info:   info,
+		Reader: nil,
+		FS:     nil,
+	}
+
+	ctx := context.Background()
+	_, err = extractor.Extract(ctx, input)
+	if err == nil {
+		t.Errorf("Extract(%q) succeeded, want error", path)
+	}
+}
+
+func TestExtractNonExistentVMDK(t *testing.T) {
+	extractor := vmdk.New()
+	path := "testdata/nonexistent.vmdk"
+	input := &filesystem.ScanInput{
+		Path:   path,
+		Root:   "testdata",
+		Info:   nil,
+		Reader: nil,
+		FS:     nil,
+	}
+
+	ctx := context.Background()
+	_, err := extractor.Extract(ctx, input)
+	if err == nil {
+		t.Errorf("Extract(%q) succeeded, want error", path)
+	}
+}
+
+type mockFileAPI struct {
```

Comment:
We already have such a util for FileRequired tests, see for example: https://github.com/google/osv-scalibr/blob/11df6d9201529dae77b6641fe18a37d080b26d84/extractor/filesystem/language/lua/luarocks/luarocks_test.go#L72